File: 1_TwoSum/twoSum.c
/**
 * Note: the returned array must be malloced, assume caller calls free()
 */

int *twoSum(int *nums, int numsSize, int target, int *returnSize)
{
    //brute force
    for(int i = 0; i < numsSize -1; i++){

        for(int j = i+1; j < numsSize; j++){

            if(nums[i] + nums[j] == target){
                *returnSize = 2;
                int *result = malloc(2 * sizeof(int));
                result[0] = i;
                result[1] = j;

                return result;
            }
        }
    }

    *returnSize = 0;

    return;
}

File: 1_TwoSum/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # n + j = target
        # target - j = n
        spam = {}
        for i in range(len(nums)):
            if nums[i] in spam:
                return [i, spam[nums[i]]]

            else:
                spam[target - nums[i]] = i